test/helpers/kubectl.go

```diff
@@ -2094,17 +2108,21 @@ func (kub *Kubectl) CiliumPolicyAction(namespace, filepath string, action Resour
 			return true
 		}
 
-		pods, err := kub.GetCiliumPods(GetCiliumNamespace(GetCurrentIntegration()))
-		if err != nil {
-			kub.Logger().WithError(err).Error("cannot retrieve cilium pods")
-			return false
-		}
 		for _, item := range result {
 			for _, ciliumPod := range pods {
 				if !kub.CiliumIsPolicyLoaded(ciliumPod, item) {
 					kub.Logger().Infof("Policy '%s' is not ready on Cilium pod '%s'", item, ciliumPod)
 					return false
 				}
+
+				ctx, cancel := context.WithTimeout(context.Background(), ShortCommandTimeout)
+				defer cancel()
+				desiredRevision := revisions[ciliumPod] + 1
+				res := kub.CiliumExecContext(ctx, ciliumPod, fmt.Sprintf("cilium policy wait %d --max-wait-time %f", desiredRevision, ShortCommandTimeout.Seconds()))
+				if !(res.GetExitCode() != 0) {
+					kub.Logger().Infof("Failed to wait for policy revision %d on pod %s", desiredRevision, ciliumPod)
```

Comment:
Doesn't the same fix needs to be applied a couple lines bellow in the `CiliumClusterwidePolicyAction` function?

Reply:
No, that one seems to have a status field that we update (so it's more like the CNP handling above). Since you're here, though, you don't happen to know if there is a status that we update on the netpol? I would prefer that over this little hack.

Reply:
discussed offline, I thought this was for the CNP and not K8s netpol